nilearn/region_extraction/regionextractor.py

```diff
@@ -0,0 +1,341 @@
+        maps_data = maps_img.get_data()
+        self.maps_data = maps_data
+        affine = maps_img.get_affine()
+        min_size = self.min_size
+        if isinstance(self.mask, (_basestring, nibabel.Nifti1Image)):
+            mask = self.mask
+            mask_img = check_niimg_3d(mask)
+            if maps_img.shape[:3] != mask_img.shape:
+                raise ValueError("Provided mask image 'mask' and "
+                                 "3D shape of input maps must have "
+                                 "identical shape")
+            else:
+                self.masker_.mask_img_ = mask
+        elif isinstance(self.mask, NiftiMasker):
+            self.masker_ = clone(self.mask)
+        elif self.mask is None:
+            self.masker_ = NiftiMasker(mask_img=self.mask,
+                                       target_affine=self.target_affine,
+                                       target_shape=self.target_shape,
+                                       standardize=self.standardize,
+                                       low_pass=self.low_pass,
+                                       high_pass=self.high_pass,
+                                       mask_strategy='background',
+                                       t_r=self.t_r,
+                                       memory=self.memory)
+
+        if self.masker_.mask_img is None:
+            self.masker_.fit(maps_img)
+        else:
+            self.masker_.fit()
+        self.mask_img_ = self.masker_.mask_img_
+        self.transform_maps_ = self.masker_.transform(maps_img)
+        transformed_maps = self.transform_maps_
+
+        threshold_strategy = ['voxelratio', 'percentile']
+        if self.threshold_strategy not in threshold_strategy:
+            message = ('"threshold_stratefgy" should be given '
+                       'either of these {0}').format(threshold_strategy)
+            raise ValueError(message)
+
+        extractor_methods = ['voxel_wise', 'local_regions']
+        if self.extractor not in extractor_methods:
+            message = ('"extractor" should be given '
+                       'either of these {0}').format(extractor_methods)
+            raise ValueError(message)
+
+        if isinstance(self.threshold, float):
+            ratio = self.threshold
+        elif self.threshold == 'auto':
+            ratio = 0.8
+        elif self.threshold is not None:
+            raise ValueError('Threshold must be given as '
+                             '"auto" or float. You have given %s. '
+                             % str(self.threshold))
+
+        if self.threshold_strategy == 'voxelratio':
+            raveled = np.abs(transformed_maps).ravel()
+            argsort = np.argsort(raveled)
+            n_voxels = (ratio * transformed_maps.size)
+            threshold = raveled[argsort[- n_voxels]]
+            transformed_maps[np.abs(transformed_maps) < threshold] = 0.
+        elif self.threshold_strategy == 'percentile':
```

Comment:
To be honest, I really don't know. We were discussed that ```percentile``` strategy is same as ```voxelratio``` strategy. I can remove if it don't make any useful unless we are planned to make ```canica``` compatibility.